src/neuroconv/datainterfaces/behavior/fictrac/fictracdatainterface.py

```diff
@@ -117,311 +120,126 @@ def add_to_nwbfile(
-            name="SpatialSeriesIntegratedMotion",
-            data=data,
-            unit="radians",
-            reference_frame=reference_frame,
-            description=description,
-        )
-
-        if write_timestamps:
-            spatial_seriess_kwargs["timestamps"] = timestamps
-        else:
-            spatial_seriess_kwargs["rate"] = rate
-            spatial_seriess_kwargs["starting_time"] = starting_time
-
-        spatial_series = SpatialSeries(**spatial_seriess_kwargs)
-        compass_direction_container.add_spatial_series(spatial_series)
-
-        # Add error in rotation delta
-        rotation_delta_error = fictrac_data_df["rotation_delta_error"].values
-        reference_frame = "lab"
-        description = "Error in rotation delta in radians from the lab frame"
-
-        spatial_seriess_kwargs = dict(
-            name="SpatialSeriesRotationDeltaError",
-            data=rotation_delta_error,
-            reference_frame=reference_frame,
-            unit="radians",
-            description=description,
-        )
-
-        if write_timestamps:
-            spatial_seriess_kwargs["timestamps"] = timestamps
-
-        else:
-            spatial_seriess_kwargs["rate"] = rate
-            spatial_seriess_kwargs["starting_time"] = starting_time
-
-        spatial_series = SpatialSeries(**spatial_seriess_kwargs)
-        compass_direction_container.add_spatial_series(spatial_series)
+        position_container = Position(name="FicTrac")
+
+        for data_dict in self.column_to_nwb_mapping.values():
+            spatial_seriess_kwargs = dict(
+                name=data_dict["spatial_series_name"],
+                description=data_dict["description"],
+                reference_frame=data_dict["reference_frame"],
+            )
+
+            columns_in_dat_file = data_dict["columns_in_dat_file"]
+            data = fictrac_data_df[columns_in_dat_file].to_numpy()
+            if self.radius is not None:
+                data = data * self.radius
+                units = "meters"
+            else:
+                units = "radians"
+
+            spatial_seriess_kwargs.update(data=data, unit=units)
+
+            if write_timestamps:
+                spatial_seriess_kwargs["timestamps"] = timestamps
+            else:
+                spatial_seriess_kwargs["rate"] = rate
+                spatial_seriess_kwargs["starting_time"] = starting_time
+
+            # Create the spatial series and add it to the container
+            spatial_series = SpatialSeries(**spatial_seriess_kwargs)
+            position_container.add_spatial_series(spatial_series)
 
         # Add the compass direction container to the processing module
         processing_module = get_module(nwbfile=nwbfile, name="Behavior")
-        processing_module.add_data_interface(compass_direction_container)
+        processing_module.add_data_interface(position_container)
+
+    column_to_nwb_mapping = spatial_series_descriptions = {
```

Comment:
?

Reply:
A class attribute with the configuration of how the data from the binary file (csv in fact) is mapped to the structure of nwb. It has columns, the name of the series the reference frame and the columns where it should be extracted from in the binary file.